### PR TITLE
Reorganize AJAX JavaScript

### PR DESCRIPTION
**Pull request by JoosuaKoskinen**
#### What are the main changes in this PR?

The JavaScript was a bit nasty so it's now
little better.

Also added cache bursting on `SERAVO_PLUGIN_DEBUG` mode and JS events
`seravoAjaxError` and `seravoAjaxRequest`.

Old code shouldn't be affected.

### CODE REVIEW

File: js/lib/ajax/ajax-handler.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  var dryrun_button = jQuery(section).find('#' + section_name + '-dryrun-button');
+  var spinner = jQuery('#' + section_name + '-spinner').closest('.seravo-spinner-wrapper');
+  var output = jQuery('#' + section_name + '-output');
+
+  function on_success(response) {
+    // Show output
+    spinner.hide();
+    output.html('<hr>' + response['output'] + '<hr>');
+    output.show();
+
```
Comment (Moppa5) [inline js/lib/ajax/ajax-handler.js:81]: Would you consider adding on this PR
Screenshot(14)

Basically adding again the optional `<hr>` tag usage as for example object cache nag looks ugly with these and of course the `.hideable` elements.

File: js/common.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+"use strict";
+
+jQuery(document).ready(
+  function () {
+
+    jQuery('.seravo-show-more-wrapper').click(on_show_more_wrapper_click);
+
```
Comment (Moppa5) [inline js/common.js:7]: Not necessarily an issue but these empty lines look weird here.

**Review by Moppa5 — COMMENTED** [review]
Looking good! I think the JS structure is now even more clearer. I had only one minor comment and some additions from `rewrite-cache-status` branch, which I would like you to consider. 

Anyway, after fixing the phpcs errors etc. I think you can feel free to merge this.

**Review by Moppa5 — APPROVED** [review]

**Comment by JoosuaKoskinen** [comment]
This has been merged in https://github.com/Seravo/seravo-plugin/commit/76f72aadb68db256b7c3b37d1370481bc2399405.

**Pull request closed without merging**